control-plane/p4RuntimeSerializer.cpp

```diff
@@ -729,6 +729,10 @@ getTypeWidth(const IR::Type* type, TypeMap* typeMap) {
         // W if the type is bit<W>, and 0 if the type is string
         return annotation.controller_type.width;
     }
+    /* Treat error type as string */
+    if (type->is<IR::Type_Error>())
+        return 0;
+
```

Comment:
I'm not reviewing the C++ implementation code parts of this PR very carefully -- simply scanning through them looking for anything odd.  Is this change something that was discovered as a small side issue while developing the fixes for preserving user-defined metadata fields for resubmit/recirculate/clone?  Perhaps something that was needed for the failing test cases exercising those features to pass, so good to combine them into this PR while you were at it?  I don't have any objection, main curious why it is thrown in with this PR.

Reply:
I think that this is coming from the main branch from a merge.